components/raftstore/src/coprocessor/mod.rs

```diff
@@ -363,6 +363,8 @@ pub trait RegionHeartbeatObserver: Coprocessor {
 }
 
 pub trait MessageObserver: Coprocessor {
+    fn on_extra_message(&self, _: &Region, _: &ExtraMessage) {}
```

Comment:
What about add a separate `ExtraMessageObserver`, not sure of observe all raft messges will hurt the performance

Reply:
Sounds good.